### PR TITLE
fix: fixes issue with webhook not being triggered for managed events

### LINKED ISSUE #9962: [CAL-2111] Managed event does not fire Webhooks if it has multiple team members
Found a bug? Please fill out the sections below. 👍

### Issue Summary

A managed event doesn't fire webhook if the it contains multiple team members. If it has a single team member, the webhooks fire.

### Steps to Reproduce

1. Create a managed event
2. Add a webhook to this event with BOOKING.CREATED trigger enabled
3. Make booking and confirm that the webhook works
4. Now add more team members to the event
5. Make another booking, and how it shouldn't trigger the webhook

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

### Technical details

- Production

<sub>[CAL-2111](https://linear.app/calcom/issue/CAL-2111/managed-event-does-not-fire-webhooks-if-it-has-multiple-team-members)</sub>

### PR DESCRIPTION
**Pull request by Pradumn27**
## What does this PR do?

This PR fixes the issue that the Booking Created webhook was not being triggered for managed events. The reason was that the eventTypes created for every assigned team member did not contain the teamId which was needed to find the right webhook while triggering.

We can't add teamId to this eventType either because [slug,teamId] is a unique identifier for the event types and it's important that it stays that way so that we can't have a team having multiple events with the same slug. So what I've done is while booking I'll check whether it's a managed event and then will extract the eventType from the eventType of this slug which contains the teamId this way we will be able to access the teamId.

Also, the event type that was being created when a team member was assigned a managed event did not contain the schedulingType so have added that too.

Fixes #9962 

https://github.com/calcom/cal.com/assets/47187878/19ffda33-0362-4fbf-82a4-7e188495c94e

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Create a managed event 
- Configure a webhook for this event with BOOKING.CREATED trigger enabled
- Add members to this event
- Make a booking
- Check that the webhook was triggered

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by CarinaWolli** [comment]
@Jaibles (or @leog, @alishaz-polymath) should managed event types trigger the user's webhooks, or should they trigger the team webhooks? 

@Pradumn27 I see in your screen share you show that it triggers for a team webhooks, however, that's not working for me 🤔

**Comment by Pradumn27** [comment]
> @Pradumn27 I see in your screen share you show that it triggers for a team webhooks, however, that's not working for me 🤔

@CarinaWolli are you using any already created user's event? Because then it will not work as I've mentioned in the pr description "The event type that was being created when a team member was assigned a managed event did not contain the schedulingType so have added that too." that might be the possible case, do let me know will debug this further then.

**Comment by CarinaWolli** [comment]
Ah I see, sorry I missed that. That was the case for me! 

We also recognize a managed event type by it having a `parent`. So we could just simply check if it has a parent and then check the parent's teamId

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 1]
```diff
@@ -2164,11 +2164,33 @@ async function handler(
...
+      },
+      select: {
+        teamId: true,
+      },
+    });
+
+    if (managedEvent?.teamId) {
+      teamId = managedEvent.teamId;
+    }
+  }
```
Comment (hariombalhara) [inline packages/features/bookings/lib/handleNewBooking.ts:2187]: Let's move this code out of `handler` function into it's own function(say getTeamId())

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 2]
```diff
@@ -2164,11 +2164,33 @@ async function handler(
...
     smsReminderNumber: booking?.smsReminderNumber || undefined,
   };
+
+  let teamId = eventType.team?.id;
+
+  // If it's a managed event we need to find the teamId for it from the originally created event with this slug
+  if (!teamId && eventType.schedulingType === SchedulingType.MANAGED) {
+    const managedEvent = await prisma.eventType.findFirst({
+      where: {
+        slug: req.body.eventTypeSlug,
```
Comment (hariombalhara) [inline packages/features/bookings/lib/handleNewBooking.ts:2174]: This is wrong way to get the parent managed event-type. Slugs can be duplicate.
We should get by parentId of the event.

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Thanks for the PR 🙏 But it needs some fixing

**Comment by Pradumn27** [comment]
@CarinaWolli @hariombalhara thanks for the review, wasn't aware of the parentId field, have made the requested adjustments please review again.

File: packages/lib/defaultEvents.ts [hunk 3]
```diff
@@ -92,6 +92,7 @@ const commons = {
   userId: 0,
   owner: null,
   workflows: [],
+  parentId: null,
```
Comment (Pradumn27) [inline packages/lib/defaultEvents.ts:95]: Added to remove the type warnings for getEventType.

File: packages/features/ee/managed-event-types/lib/handleChildrenEventTypes.ts [hunk 4]
```diff
@@ -205,6 +205,7 @@ export default async function handleChildrenEventTypes({
               },
             },*/
             hashedLink: hashedLinkQuery(userId),
+            schedulingType: updatedEventType.schedulingType,
```
Comment (hariombalhara) [inline packages/features/ee/managed-event-types/lib/handleChildrenEventTypes.ts:208]: How is this newly added schedulingType property being used ?

**Comment by hariombalhara** [comment]
> > @Pradumn27 I see in your screen share you show that it triggers for a team webhooks, however, that's not working for me 🤔
> 
> @CarinaWolli are you using any already created user's event? Because then it will not work as I've mentioned in the pr description "The event type that was being created when a team member was assigned a managed event did not contain the schedulingType so have added that too." that might be the possible case, do let me know will debug this further then.

Do we need schedulingType? The requirement is to simply get the parent team ID which is already available in parentId. My concern is to to get it working for existing event-types as well

File: packages/features/ee/managed-event-types/lib/handleChildrenEventTypes.ts (hunk 4 above)
Comment (Pradumn27) [reply packages/features/ee/managed-event-types/lib/handleChildrenEventTypes.ts:208]: It isn't used currently, but for database consistency shouldn't this event still have this schedulingType?

**Comment by Pradumn27** [comment]
> Do we need schedulingType? The requirement is to simply get the parent team ID which is already available in parentId. My concern is to to get it working for existing event-types as well

@hariombalhara As per the current changes it will work for the existing event-types as well, in the case of schedulingType I figured if we want to have database consistency shouldn't these managed events also have this schedulingType?

**Review by hariombalhara — APPROVED** [review]
Awesome work @Pradumn27 🙏 Thank you !!

**Pull request merged**